ecommerce/extensions/payment/processors/braintree.py

```diff
@@ -0,0 +1,224 @@
+""" Braintree payment processing. """
+from __future__ import absolute_import, unicode_literals
+import logging
+from urlparse import urljoin
+
+import braintree
+from braintree.attribute_getter import AttributeGetter
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from oscar.apps.payment.exceptions import GatewayError
+from oscar.core.loading import get_model
+
+from ecommerce.extensions.order.constants import PaymentEventTypeName
+from ecommerce.extensions.payment.processors import BasePaymentProcessor
+
+logger = logging.getLogger(__name__)
+
+PaymentEvent = get_model('order', 'PaymentEvent')
+PaymentEventType = get_model('order', 'PaymentEventType')
+PaymentProcessorResponse = get_model('payment', 'PaymentProcessorResponse')
+ProductClass = get_model('catalogue', 'ProductClass')
+Source = get_model('payment', 'Source')
+SourceType = get_model('payment', 'SourceType')
+
+
+class Braintree(BasePaymentProcessor):
+    NAME = 'braintree'
+
+    def __init__(self):
+        """
+        Constructs a new instance of the Braintree processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor
+            AttributeError: If ECOMMERCE_URL_ROOT setting is not set.
+        """
+        configuration = self.configuration
+        self.enviroment_mode = configuration['mode']
+        self.merchant_id = configuration['merchant_id']
+        self.public_key = configuration['public_key']
+        self.private_key = configuration['private_key']
+        self.receipt_page_url = configuration['receipt_page_url']
+        self.ecommerce_url_root = settings.ECOMMERCE_URL_ROOT
+
+        if self.enviroment_mode == 'sandbox':
```

Comment:
Nitpick: I would always prefer checks for == 'production'. This way if you make a typo in the code somewhere you will get to production mode by mistake, when it would be more secure the other way round :)